src/agent.cpp

```diff
@@ -249,62 +277,68 @@ void Agent::actionImpl() {
   }
 
   // For now let's not worry about turning the neck or setting the vision.
-  // However, do this now so doesn't override anything changed by the requested action.
-  // TODO for librcsc: setViewActionDefault, setNeckActionDefault that will not overwrite if already set.
+  // But do the settings now, so that doesn't override any set by the actions below.
+  // TODO: Add setViewActionDefault, setNeckActionDefault to librcsc that only set if not already set.
   this->setViewAction(new View_Tactical());
   this->setNeckAction(new Neck_TurnToBallOrScan());
 
+  const WorldModel & wm = this->world();
+  bool may_fix = wm.self().collidesWithPost();
```

Comment:
In response to the first question, yes. In response to the second question, some of the methods return a false boolean return even when they succeed in fixing this situation (something that does not tend to happen nearly as frequently with player or ball collisions, probably because of a different collision implementation). Various revisions of the programs I used for this can be seen in the docs_update branch history, plus the current (albeit needing corrections for distance changes) in the explore_high_from_low branch.